docs/Dialects/FIRRTL/FIRRTLIntrinsics.md

```diff
@@ -94,3 +94,108 @@ The enable input is sampled at the rising edge of the input clock; any changes o
 | in   | input     | Clock    | input clock                 |
 | en   | input     | UInt<1>  | enable for the output clock |
 | out  | output    | Clock    | gated output clock          |
+
+### circt.chisel_assert_assume
+
+Generate a clocked SV assertion with companion assume statement.
+
+Has legacy special behavior and should not be used by new code.
+
+| Parameter | Type   | Description                                                                         |
+| --------- | ------ | ----------------------------------------------------------------------------------- |
+| format    | string | Format string per SV 20.10, 21.2.1.  Optional.                                      |
+| label     | string | Label for assert/assume.  Optional.                                                 |
+| guards    | string | Semicolon-delimited list of pre-processor tokens to use as ifdef guards.  Optional. |
+
+| Port      | Direction | Type     | Description                |
+| --------- | --------- | -------- | -------------------------- |
+| clock     | input     | Clock    | input clock                |
+| predicate | input     | UInt<1>  | predicate to assert/assume |
+| enable    | input     | UInt<1>  | enable signal              |
+| ...       | input     | Signals  | arguments to format string |
+
+Example output:
+```systemverilog
+wire _GEN = ~enable | cond;
+assert__label: assert property (@(posedge clock) _GEN) else $error("message");
+`ifdef USE_PROPERTY_AS_CONSTRAINT
+  assume__label: assume property (@(posedge clock) _GEN);
```

Comment:
I guess _GEN is used as edge (maybe I'm wrong). 
```suggestion
  assume__label: assume property (@(edge _GEN) _GEN);
```

Reply:
Thank you, this is what is produced "normally" -- the at-edge "predicate" style is one of the "legacy special" behaviors this becomes, specifically when a certain guard string is present, the companion assume instead of the above is:

```systemverilog
  `ifdef USE_UNR_ONLY_CONSTRAINTS	// <stdin>:3:5
    wire _GEN = ~enable | cond;	// <stdin>:3:5
    assert__label: assert property (@(posedge clock) _GEN);	// <stdin>:3:5
    `ifdef USE_PROPERTY_AS_CONSTRAINT	// <stdin>:3:5
      always @(edge _GEN)	// <stdin>:3:5
        assume__label: assume(_GEN);	// <stdin>:3:5
    `endif // USE_PROPERTY_AS_CONSTRAINT
  `endif // USE_UNR_ONLY_CONSTRAINTS
```

So I think this is "okay" as-is but happy to change it in a follow-up.

Thanks for taking the time and suggestions/attention!!